server/enterprise/message_export/message_export_test_e2e_generators.go

```diff
@@ -110,10 +208,154 @@ type SetupReturn struct {
 	channels          []*model.Channel
 	teams             []*model.Team
 	batchTimes        []batchStartEndTimes
+	jobExportDir      string
+}
+
+func generateActianceBatchTest1(t *testing.T, th *api4.TestHelper, attachmentDir, exportDir string,
+	attachmentBackend filestore.FileBackend) JobResults {
+	waitUntilZeroPosts(t, th)
+	now := model.GetMillis()
+	jobStart := now - 1
+
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.MessageExportSettings.EnableExport = true
+		*cfg.MessageExportSettings.ExportFromTimestamp = jobStart
+		*cfg.MessageExportSettings.BatchSize = 5
+		*cfg.MessageExportSettings.ExportFormat = model.ComplianceExportTypeActiance
+		*cfg.FileSettings.DriverName = model.ImageDriverLocal
+		*cfg.FileSettings.Directory = attachmentDir
+
+		if exportDir != attachmentDir {
+			*cfg.FileSettings.DedicatedExportStore = true
+			*cfg.FileSettings.ExportDriverName = model.ImageDriverLocal
+			*cfg.FileSettings.ExportDirectory = exportDir
+		}
+	})
+
+	contents := []string{"Hello there"}
+	attachmentPath001 := "path/to/attachments/one.txt"
+	_, _ = attachmentBackend.WriteFile(bytes.NewBufferString(contents[0]), attachmentPath001)
+	post, err := th.App.Srv().Store().Post().Save(th.Context, &model.Post{
+		ChannelId: th.BasicChannel.Id,
+		UserId:    model.NewId(),
+		Message:   "zz" + model.NewId() + "b",
+		CreateAt:  now,
+		UpdateAt:  now,
+		FileIds:   []string{"test1"},
+	})
+	require.NoError(t, err)
+
+	attachment, err := th.App.Srv().Store().FileInfo().Save(th.Context, &model.FileInfo{
+		Id:        model.NewId(),
+		CreatorId: post.UserId,
+		PostId:    post.Id,
+		CreateAt:  now,
+		UpdateAt:  now,
+		Path:      attachmentPath001,
+	})
+	require.NoError(t, err)
+	attachments := []*model.FileInfo{attachment}
+
+	for i := 0; i < 10; i++ {
+		_, e := th.App.Srv().Store().Post().Save(th.Context, &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			UserId:    model.NewId(),
+			Message:   "zz" + model.NewId() + "b",
+			CreateAt:  now + int64(i),
+			UpdateAt:  now + int64(i),
+		})
+		require.NoError(t, e)
+	}
+
+	until := model.GetMillis()
```

Comment:
Should we worry about this ever matching `now`? Wondering if it'd be safer to calculate this relative to the initial value.

Reply:
So far it hasn't, and I've tested a lot. The reason I wouldn't calculate, is because those `start` and `until` values get sent to the worker (through the db in json) and we'll need those to both be unix milliseconds wall clock time. If we ever have the case where it starts equalling, I think the only solution is to sleep more, really... I don't know. We're operating with so many clock times (test's clock, worker's clock, server's clock when it inserts posts/fileinfos/etc.)... There's no authoritative wall clock (even though it's all the same process, and even the same line of code isn't authoritative...)  Anyway, that's my thinking behind keeping it as is. Let me know if you have thoughts or disagree.